aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1204,9 +1206,11 @@ Tensor& logsumexp_out(const Tensor& self, IntArrayRef dims, bool keepdim, Tensor
 }
 
 Tensor logsumexp(const Tensor& self, IntArrayRef dims, bool keepdim) {
-  Tensor result = at::empty({0}, self.options());
+  Tensor result;
```

Comment:
use `at::promote_types(self.dtype(), kFloat)` to get the result dtype.

Reply:
Addressed!